File: baekjoon/Kotlin/2559_수열.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))

    val (N, K) = br.readLine().split(" ").map { it.toInt() }

    val lst = br.readLine().split(" ").map { it.toInt() }.toIntArray()

    var sm = lst.take(K).sum()

    var ans = sm

    for (i in K until N) {
        sm = sm + lst[i] - lst[i - K]
        ans = maxOf(ans, sm)
    }
    println(ans)
}

File: baekjoon/Python/2559_수열.py
N, K = map(int, input().split())

lst = list(map(int, input().split()))

ans = sm = sum(lst[:K])
for i in range(K, N):
    # 바로 뒷자리를 추가, 제일 앞을 제거해서 연산 최소화
    sm = sm + lst[i] - lst[i - K]
    ans = max(ans, sm)
print(ans)
